### PR TITLE
[Do Not Merge] Kubernetes Executor Rebased (candidate)

### PR DESCRIPTION
**Pull request by dimberman**
Make sure you have checked _all_ steps below.

### JIRA
- [ ] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-XXX

### Description
- [ ] Here are some details about my PR, including screenshots of any UI changes:

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/contrib/kubernetes/kubernetes_request_factory/pod_request_factory.py [hunk 1]
```diff
@@ -33,6 +34,9 @@ class SimplePodRequestFactory(KubernetesRequestFactory):
     - name: base
       image: airflow-slave:latest
       command: ["/usr/local/airflow/entrypoint.sh", "/bin/bash sleep 25"]
+      volumeMounts:
+        - name: shared-data
+          mountPath: "/usr/local/airflow/dags"
```
Comment (bolkedebruin) [inline airflow/contrib/kubernetes/kubernetes_request_factory/pod_request_factory.py:39]: /var/lib is more standard

**Comment by bolkedebruin** [comment]
Any updates? We are getting slowly to a 1.10 release and I know some ppl are really eager to try this out, like @Fokko .

**Comment by dimberman** [comment]
@bolkedebruin I've recently been focusing on the Kubernetes Operator article (meeting with bloomberg's in-house tech writer with a completed first draft this Wednesday) so that should be ready for release soon.

WRT the executor, it's working on my local machine but it seems there are some minor issues with the travis build, so I'm going to investigate that tonight (hopefully it's a straightforward fix), but should have that fixed this week. If I can get the build fixed on travis, I would be glad to host a Q&A session to help early adopters get set up (or even just ask people to join kubernetes.slack.com where I can asynchronously answer questions).

**Comment by dimberman** [comment]
Please refer to https://github.com/apache/incubator-airflow/pull/2414

**Pull request closed without merging**